src/blockchain/compact_filters/mod.rs

```diff
@@ -580,3 +580,22 @@ impl From<crate::error::Error> for CompactFiltersError {
         CompactFiltersError::Global(Box::new(err))
     }
 }
+
+#[cfg(test)]
+#[cfg(feature = "test-compact-filters")]
+mod test {
+
+    use super::*;
+
+    crate::bdk_blockchain_tests! {
+        fn test_instance(test_client: &TestClient) -> CompactFiltersBlockchain {
+            let bitcoind_port = test_client.bitcoind.params.rpc_socket.port();
+            let num_threads = 2;
+            let mempool = Arc::new(Mempool::default());
+            let peers = (0..num_threads)
+            .map(|_| Peer::connect(format!("localhost:{}", bitcoind_port), Arc::clone(&mempool), Network::Regtest))
+            .collect::<Result<_, _>>().unwrap();
+            CompactFiltersBlockchain::new(peers, "./wallet-filters", Some(0)).unwrap()
```

Comment:
I haven't tested yet, but I don't think this logic will work.. 

 - You can't get compact filters from rpc port. You have to connect to the p2p port.
 - The test client doesn't serve compact filters by default. You need to set a extra bitcoind args parameter accordingly to turn that on the test client.. Which can be done under a feature gate..